core/src/main/java/org/kiwix/kiwixmobile/core/main/AddNoteDialog.kt

```diff
@@ -0,0 +1,444 @@
+      add_note_edit_text.text.clear()
+      disableMenuItems()
+      context.toast(R.string.note_delete_successful, Toast.LENGTH_LONG)
+    } else {
+      context.toast(R.string.note_delete_unsuccessful, Toast.LENGTH_LONG)
+    }
+  }
+
+  /* String content of the note text file given at:
+   *    "{External Storage}/Kiwix/Notes/ZimFileTitle/ArticleTitle.txt"
+   * is displayed in the EditText field (note content area)
+   */
+  private fun displayNote() {
+
+    val noteFile = File("$zimNotesDirectory$articleNoteFileName.txt")
+    if (noteFile.exists()) {
+      writeNote(noteFile)
+    }
+
+    // No action in case the note file for the currently open article doesn't exist
+  }
+
+  private fun writeNote(noteFile: File) {
+    noteFileExists = true
+    val contents = noteFile.readText()
+    add_note_edit_text.setText(contents) // Display the note content
+    add_note_edit_text.setSelection(add_note_edit_text.text.length - 1)
+    enableShareNoteMenuItem() // As note content exists which can be shared
+    enableDeleteNoteMenuItem()
+  }
+
+  private fun shareNote() {
+
+    /* The note text file corresponding to the currently open article, given at:
+     *    "{External Storage}/Kiwix/Notes/ZimFileTitle/ArticleTitle.txt"
+     * is shared via an app-chooser intent
+     * */
+    if (noteEdited) {
+      // Save edited note before sharing the text file
+      saveNote(add_note_edit_text.text.toString())
+    }
+    val noteFile = File("$zimNotesDirectory$articleNoteFileName.txt")
+    if (noteFile.exists()) {
+      val noteFileUri = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+        // From Nougat 7 (API 24) access to files is shared temporarily with other apps
+        // Need to use FileProvider for the same
+        FileProvider.getUriForFile(
+          requireContext(), requireContext().packageName + ".fileprovider",
+          noteFile
+        )
+      } else {
+        Uri.fromFile(noteFile)
+      }
+      val noteFileShareIntent = Intent(Intent.ACTION_SEND)
+      noteFileShareIntent.type = "application/octet-stream"
+      noteFileShareIntent.putExtra(Intent.EXTRA_STREAM, noteFileUri)
+      noteFileShareIntent.flags = Intent.FLAG_GRANT_READ_URI_PERMISSION
+      val shareChooser = Intent.createChooser(
+        noteFileShareIntent,
+        getString(R.string.note_share_app_chooser_title)
+      )
+      if (noteFileShareIntent.resolveActivity(requireActivity().packageManager) != null) {
+        startActivity(shareChooser)
+      }
+    } else {
+      context.toast(R.string.note_share_error_file_missing, Toast.LENGTH_SHORT)
+    }
+  }
+
+  private fun dismissAddNoteDialog() {
+    val dialog = dialog
+    dialog?.dismiss()
+    closeKeyboard()
+  }
+
+  override fun onStart() {
+    super.onStart()
+    dialog?.let {
+      it.window?.setLayout(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT)
+    }
+  }
+
+  companion object {
+    @JvmField val NOTES_DIRECTORY =
```

Comment:
Why `@JvmField`?

Reply:
this is used on companion objects so when it is accessed from Java it is still accessed like a static variable, without it it would be `AddNoteDialog.Companion.NOTES_DIRECTORY` that said I am not 100% this is accessed via java code

Reply:
I was under the impression that this job is done via `@JvmStatic` annotation, but this does work functions